script/subtitles/imscsubtitles.js

```diff
@@ -4,28 +4,106 @@ define('bigscreenplayer/subtitles/imscsubtitles',
     'bigscreenplayer/domhelpers',
     'bigscreenplayer/debugger/debugtool',
     'bigscreenplayer/plugins',
-    'bigscreenplayer/utils/playbackutils'
+    'bigscreenplayer/utils/playbackutils',
+    'bigscreenplayer/utils/loadurl'
   ],
-  function (IMSC, DOMHelpers, DebugTool, Plugins, Utils) {
+  function (IMSC, DOMHelpers, DebugTool, Plugins, Utils, LoadURL) {
     'use strict';
-    return function (mediaPlayer, response, autoStart, parentElement, defaultStyleOpts) {
+    return function (mediaPlayer, captions, autoStart, parentElement, defaultStyleOpts, windowStartTime) {
       var currentSubtitlesElement;
       var exampleSubtitlesElement;
-      var previousSubtitlesIndex = null;
       var imscRenderOpts = transformStyleOptions(defaultStyleOpts);
       var updateInterval;
-      var xml;
-      var times = [];
+      var SEGMENTS_TO_KEEP = 3;
+      var segments = [];
+      var currentSegmentRendered = {};
+      var liveSubtitles = !!captions.segmentLength;
 
-      try {
-        xml = IMSC.fromXML(response.text);
-        times = xml.getMediaTimeEvents();
-        if (autoStart) {
-          start();
+      if (autoStart) {
+        start();
+      }
+
+      function loadAllRequiredSegments () {
+        var segmentsToLoad = [];
+
+        for (var i = 0; i < SEGMENTS_TO_KEEP; i++) {
+          var segmentNumber = calculateSegmentNumber(i);
+          var alreadyLoaded = segments.some(function (segment) {
+            return segment.number === segmentNumber;
+          });
+
+          if (!alreadyLoaded) {
+            segmentsToLoad.push(segmentNumber);
+          }
+        }
+
+        if (SEGMENTS_TO_KEEP === segmentsToLoad.length) {
+          // This is to ensure when seeking to a point with no subtitles, don't leave previous subtitle displayed.
+          removeCurrentSubtitlesElement();
+        }
+
+        segmentsToLoad.forEach(function (segmentNumber) {
+          loadSegment(captions.captionsUrl, segmentNumber);
+        });
+      }
+
+      function calculateSegmentNumber (offset) {
+        var epochSeconds = (windowStartTime / 1000) + mediaPlayer.getCurrentTime();
+        return Math.floor(epochSeconds / captions.segmentLength) + offset;
+      }
+
+      function loadSegment (url, segmentNumber) {
+        url = url.replace('$segment$', segmentNumber);
+        LoadURL(url, {
+          onLoad: function (responseXML, responseText, status) {
+            if (!responseXML && !liveSubtitles) {
+              DebugTool.info('Error: responseXML is invalid.');
+              Plugins.interface.onSubtitlesTransformError();
+              stop();
+              return;
+            }
+
+            try {
+              var xml = IMSC.fromXML(responseText.split(/<\?xml version=\"1.0\" encoding=\"UTF-8\"\?>/i)[1] || responseText);
```

Comment:
Live subtitles contain more than just the XML in the response which is why we need to split. The XML header which we split on isn't needed to be considered valid XML.

Reply:
My gut feeling says we shouldn't mutate the XML object by stripping the head off. 

However if IMSC accepts it then it's likely fine. Hoping we don't get tripped up by this in the future though if we were to use the XML for something else further in the code.

The reason I say this is because removing the encoding declaration _could_ be an issue. But I imagine parsers autodetect this stuff so it should be fine.